dashboard-app/src/dashboard/ByTime/helper.ts

```diff
@@ -0,0 +1,60 @@
+import { Duration } from 'twine-util';
+import { mergeAll } from 'ramda';
+import moment from 'moment';
+import { DataTableRow, DataTableProps } from '../../components/DataTable/types';
+import { DurationUnitEnum } from '../../types';
+import { roundToDecimal, addDecimals } from '../../util/mathUtil';
+
+interface Params { data: any[]; months: string[]; unit: DurationUnitEnum; }
+
+const toUnitDuration = (unit: DurationUnitEnum, duration: Duration.Duration) => {
+  switch (unit){
+    case DurationUnitEnum.DAYS:
+      return roundToDecimal(Duration.toWorkingDays(duration));
+
+    case DurationUnitEnum.HOURS:
+      return roundToDecimal(Duration.toHours(duration));
+  }
+};
+
+const addDurationToTableContents = (
+  row: DataTableRow,
+  columnKey: string,
+  unit: DurationUnitEnum,
+  duration: Duration.Duration) =>
+    addDecimals(Number(row.columns[columnKey].content), toUnitDuration(unit, duration));
+
+export const timeLogsToTable = ({ data, months, unit }: Params): DataTableProps => {
```

Comment:
This is all a little difficult to follow. 

I don't think we should necessarily address it right now, but just as a general comment, data transformations tend to be easier to write/read/maintain when they are separated out into stages. I think there's a fairly clear conceptual distinction between _aggregation_ (calculating totals, turning a `VolunteerLog[]` into something that can be indexed by month/activity/volunteer/whatever), and _presentation_ (changing values from days to hours, setting the precision, etc). 

This also allows you to choose data structures for each stage that are most suited to the operations that you have to do. The `DataTable` expects `{ rows: { columns: Dictionary<...> }[] }` because that makes sense for rendering, but it doesn't necessarily make sense for whatever calculations need to be performed.